src/restapi/middleware.ts

```diff
@@ -0,0 +1,38 @@
+import { Request, Response } from 'express';
+
+import { ExceptionType } from '../constant';
+import { AppActingUser, AppCallRequest, Oauth2App } from '../types';
+import { Exception } from '../utils/exception';
+import { configureI18n } from '../utils/translations';
+import { isConnected, isUserSystemAdmin, showMessageToMattermost } from '../utils/utils';
+
+export const requireSystemAdmin = (req: Request, res: Response, next: () => void) => {
+    const call: AppCallRequest = req.body as AppCallRequest;
+    const i18nObj = configureI18n(call.context);
+    const actingUser: AppActingUser = call.context.acting_user as AppActingUser;
+
+    if (!actingUser) {
+        res.json(showMessageToMattermost(new Exception(ExceptionType.TEXT_ERROR, i18nObj.__('general.validation-user.not-provided'), i18nObj.__('general.validation-user.not-provided'), call)));
+        return;
+    }
+
+    if (!isUserSystemAdmin(actingUser)) {
+        res.json(showMessageToMattermost(new Exception(ExceptionType.TEXT_ERROR, i18nObj.__('general.validation-user.system-admin'), i18nObj.__('general.validation-user.system-admin'), call)));
+        return;
+    }
+
+    next();
+};
+
+export const requireUserOAuthConnected = (req: Request, res: Response, next: () => void) => {
+    const call: AppCallRequest = req.body as AppCallRequest;
+    const i18nObj = configureI18n(call.context);
+    const oauth2: Oauth2App = call.context.oauth2 as Oauth2App;
```

Comment:
Let's avoid the `as ...` in these middleware functions for the individual fields in the HTTP body. We should assume this could be undefined, and we should let the typescript compiler take care of ensuring we check that it's defined.

```suggestion
    const oauth2: Oauth2App | undefined = call.context.oauth2;
```

Reply:
This was fixed in commit [77c8af7](https://github.com/liz-segura98/mattermost-app-pagerduty/pull/1/commits/77c8af7e187a49c7fd4f2ce3f808a9181071ba55)